### ISSUE TITLE
Why the result I train can only reach the loss

### ISSUE THREAD

**Issue by lvqiao490656285** [kaituoxu/Conv-TasNet#8]

**Comment by kaituoxu** [comment]
em...I'm not sure.
Did you use the default config in egs/wjs0/run.sh?

**Comment by tky1117** [comment]
I also have the similar results; SDRi is 13.49, SI-SNRi is 12.94.
And separated sources seem to be so distorted compared to audio samples by the author.

**Comment by adiyoss** [comment]
I also got similar results. Couldn't improve above ~13.4 SDRi.
Regarding the distorted samples you should set norm =True in the librosa write function.

**Comment by tky1117** [comment]
@adiyoss Thanks for your advice. I didn't notice comment out.

**Comment by Satishpas2** [comment]
> I also got similar results. Couldn't improve above ~13.4 SDRi.
> Regarding the distorted samples you should set norm =True in the librosa write function.

thanks a lot.Without norm=True the separated files are a useless.

**Comment by lvqiao490656285** [comment]
> em...I'm not sure.
> Did you use the default config in egs/wjs0/run.sh?

yes,every config is set as your file egs/wjs0/run.sh

**Comment by enk100** [comment]
I also have similar results.
Training the model with the same configuration as in egs/wjs0/run.sh. The WSJ0 dataset obtain with the same manner in tools folder.
Training & evaluation loss looks exactly the as the loss.png

but the SDRi/SI-SDRi is poor:
SDRi is 12.49, SI-SNRi is 12.54.

**Comment by deciding** [comment]
> Training the model with the same configuration as in egs/wjs0/run.sh. The WSJ0 dataset obtain with the same manner in tools folder.
> Training & evaluation loss looks exactly the as the loss.png
> 
> but the SDRi/SI-SDRi is poor:
> SDRi is 12.49, SI-SNRi is 12.54.

Hi, is your SI-SNRi tested on the data/tt? mine can only get SISNRi around 10, even with author's trained model. but the training loss and cv loss are the same.

**Comment by lvqiao490656285** [comment]
&nbsp;yes my test set is on data/tt .SI-SNRi is &nbsp;around 12.00 using author's trained model ''final.pth.tar'' which is posted on his github. If I use the 100 epoch result model trained by myself, it only got SI-SNRi 12.51.&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"deciding"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月29日(星期天) 下午4:53
收件人:&nbsp;"kaituoxu/Conv-TasNet"<Conv-TasNet@noreply.github.com&gt;;
抄送:&nbsp;"490656285"<490656285@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [kaituoxu/Conv-TasNet] Why the result I train can only reach the loss:( -14 valid and -16 train), the evaluate result is not 15.45 too (#8)



 
Training the model with the same configuration as in egs/wjs0/run.sh. The WSJ0 dataset obtain with the same manner in tools folder.
 Training &amp; evaluation loss looks exactly the as the loss.png
 
but the SDRi/SI-SDRi is poor:
 SDRi is 12.49, SI-SNRi is 12.54.
  
Hi, is your SI-SNRi tested on the data/tt? mine can only get SISNRi around 10, even with author's trained model. but the training loss and cv loss are the same.

**Comment by deciding** [comment]
> &nbsp;yes my test set is on data/tt .SI-SNRi is &nbsp;around 12.00 using author's trained model ''final.pth.tar'' which is posted on his github. If I use the 100 epoch result model trained by myself, it only got SI-SNRi 12.51.&nbsp;
> […](#)
Thanks for the reply! Can I have a look at your conv-tasnet `pip freeze`, I think maybe the package versions cause the problem.

**Comment by lvqiao490656285** [comment]
python 3.5&nbsp; librosa 0.7.1 torch 0.4.1






------------------&nbsp;原始邮件&nbsp;------------------
发件人: "deciding"<notifications@github.com&gt;; 
发送时间: 2019年12月30日(星期一) 上午10:38
收件人: "kaituoxu/Conv-TasNet"<Conv-TasNet@noreply.github.com&gt;; 
抄送: "490656285"<490656285@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [kaituoxu/Conv-TasNet] Why the result I train can only reach the loss:( -14 valid and -16 train), the evaluate result is not 15.45 too (#8)



 
&nbsp;yes my test set is on data/tt .SI-SNRi is &nbsp;around 12.00 using author's trained model ''final.pth.tar'' which is posted on his github. If I use the 100 epoch result model trained by myself, it only got SI-SNRi 12.51.&nbsp;
 …
 Thanks for the reply! Can I have a look at your conv-tasnet pip freeze, I think maybe the package versions cause the problem.
